src/types/observablemap.ts

```diff
@@ -320,45 +322,40 @@ export class ObservableMap<K = any, V = any>
     clear() {
         transaction(() => {
             untracked(() => {
-                this._keys.slice().forEach(key => this.delete(key))
+                for (const key of this.keys()) this.delete(key)
             })
         })
     }
 
     replace(values: ObservableMap<K, V> | IKeyValueMap<V> | any): ObservableMap<K, V> {
         transaction(() => {
             const replacementMap = convertToMap(values)
-            const oldKeys = this._keys
+            const orderedData = new Map()
+            const oldKeys = Array.from(this.keys())
```

Comment:
See previous comment, IE doesn't have Array.from, so that would be problematic for mobx 4